lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -1167,8 +1167,8 @@ def main():
                 module.exit_json(changed=True)
             (rc, out, err)=nmcli.down_connection()
             (rc, out, err)=nmcli.remove_connection()
-        if rc!=0:
-            module.fail_json(name =('No Connection named %s exists' % nmcli.conn_name), msg=err, rc=rc)
+            if rc!=0:
+                module.fail_json(name =('No Connection named %s exists' % nmcli.conn_name), msg=err, rc=rc)
```

Comment:
Doesn't this mean you still return an error if the connection does not exists but write to stdout? For my understanding if a connection does not exists and `state=absent` is requested the module is done. No reason to raise an error if something is absent that is supposed to be absent anyhow.